AK/StringImpl.h

```diff
@@ -60,7 +60,7 @@ class StringImpl : public RefCounted<StringImpl> {
     {
         if (length() != other.length())
             return false;
-        return !__builtin_memcmp(characters(), other.characters(), length());
+        return (0 == __builtin_memcmp(characters(), other.characters(), length()));
```

Comment:
Does clang-tidy require yoda conditions? If not, could we not use them?

Reply:
Nah these were manually added without regard for consistency. I can make them consistent and non-Yoda. But for some cases if there's a huge expression we're comparing vs 0 it seems more readable to me to put the 0 up front 🤷‍♂️